docs/reference/developer/adr/0017-restrict-dataset-to-items-with-matching-dimensionality.rst

```diff
@@ -0,0 +1,98 @@
+ADR 0017: Restrict Dataset to items with matching dimensionality
+================================================================
+
+- Status: proposed
+- Deciders: Jan-Lukas, Neil, Simon, Sunyoung
+- Date: 2023-07-04
+
+Context
+-------
+
+Background
+~~~~~~~~~~
+
+``Dataset`` was conceived as a way to represent a collection of items with compatible dimension extents, but possibly different sets or subsets of dimensions.
+Recently we have introduced ``DataGroup``, which drops the restriction of compatible dimension extents but, unlike ``Dataset``, does in turn not provide support for joint coordinates.
+The addition of ``DataGroup`` was triggered by a long series of cases where we realized that ``Dataset`` is not useful and flexbile enough in real applications.
+This is not to say that ``Dataset`` is entirely useless, but it is not as useful as we had hoped.
+One key area that is not covered by ``DataGroup`` is the representation of table-like data (or multi-dimensional generalizations thereof), in a manner similar to ``pandas.DataFrame``.
```

Comment:
Since the name `Dataset` was inspired by the Xarray `Dataset` which can have members with different dimensionality, should we consider renaming our `Dataset` to `DataFrame`?

Reply:
I actually had a paragraph with that consideration in an earlier draft, and then removed it :D

I now think that `DataFrame` is too loaded by Pandas knowledge:

- Always 1-D
- Indices, no coordinates
- No dimension labels
- Big well known API that we do not plan to adopt